buildkite/bazelci.py

```diff
@@ -873,6 +875,54 @@ def downstream_projects_root(platform):
     return downstream_root
 
 
+def match_matrix_attr_pattern(s):
+    return re.match("^\${{\s*(\w+)\s*}}$", s)
+
+
+def get_matrix_attributes(task):
+    """Get unexpaneded matrix attributes from the given task.
+
+    If a value of field matches "${{<name>}}", then <name> is a wanted matrix attribute.
+    eg. platform: ${{ platform }}
+    """
+    attributes = []
+    for key, value in task.items():
+        if type(value) is str:
+            res = match_matrix_attr_pattern(value)
+            if res:
+                attributes.append(res.groups()[0])
+    return list(set(attributes))
+
+
+def get_combinations(matrix, attributes):
+    """Given a matrix and the wanted attributes, return all possible combinations.
+
+    eg.
+    With matrix = {'a': [1, 2], 'b': [1], 'c': [2]},
+    if attributes = ['a', 'b'], then returns [[('a', 1), ('b', 1)], [('a', 2), ('b', 1)]]
+    if attributes = ['b', 'c'], then returns [[('b', 1), ('c', 1)]]
+    if attributes = ['c'], then returns [[('c', 1)]]
```

Comment:
"then returns [[('c', **2**)]]"?

Reply:
Nice catch, but I'll fix the matrix instead.

Reply:
Also rebase with master branch after merging https://github.com/bazelbuild/continuous-integration/pull/1300